display_list/dl_op_records.h

```diff
@@ -806,14 +807,14 @@ struct DrawVerticesOp final : DrawOpBase {
   struct name##Op final : DrawOpBase {                                   \
     static const auto kType = DisplayListOpType::k##name;                \
                                                                          \
-    name##Op(const sk_sp<DlImage> image,                                 \
+    name##Op(const sk_sp<DlImage>& image,                                \
              const SkPoint& point,                                       \
              DlImageSampling sampling)                                   \
         : point(point), sampling(sampling), image(std::move(image)) {}   \
                                                                          \
     const SkPoint point;                                                 \
     const DlImageSampling sampling;                                      \
-    const sk_sp<DlImage> image;                                          \
+    const sk_sp<DlImage>& image;                                         \
```

Comment:
This seems like a bad fix. We need to copy the image. Otherwise we have a dangling reference.

Not sure if this is the cause of the test failures. At the very least, the old code isn't doing what the new code is.

Reply:
This was just reverted locally, yes this is a dangling failure!